conans/client/downloaders/caching_file_downloader.py

```diff
@@ -63,6 +63,9 @@ def _caching_download(self, urls, file_path,
             else:
                 with set_dirty_context_manager(cached_path):
                     for backup_url in backups_urls:
+                        if backup_url is None:
+                            raise ConanException("Trying to download sources from None backup remote"
+                                                 f". Remotes were: {backups_urls}")
```

Comment:
If we're going to raise an exception, should it not be better to put this condition outside of the loop?